### ISSUE TITLE
Segfault polling for XInput RawMotionEvent

### ISSUE THREAD

**Issue by jhgarner** [rust-x-bindings/rust-xcb#243]
Using the following code and xcb built with the xinput feature results in a segfault:

```rust
fn main() {
    let (main, _) = xcb::Connection::connect(Some(":0")).unwrap();
    loop {
        let event = main.register_for_special_xge::<xcb::xinput::RawMotionEvent>();
        main.poll_for_special_event(event).unwrap();
    }
}
```

Is there some other way to get RawMotionEvents from xcb?

**Comment by rtbo** [comment]
I confirm it also crashes for me

**Comment by rtbo** [comment]
After some digging, my API for special events appears to be incorrectly designed.
Here is an [online example](https://chromium.googlesource.com/chromiumos/third_party/mesa/+/7fd6ad7adcf8a1c0fc1b2746d47ba5f05b8a3c7f/src/glx/dri3_glx.c) of using the C API of special event.
My API simply doesn't allow to do it and needs to be rewritten.

In regards to the XInput RawMotionEvent, I'm not sure if special event is the correct way to go though.
I tried to adapt the [xinput_stylus_event example](https://github.com/rust-x-bindings/rust-xcb/blob/main/examples/xinput_stylus_events.rs), but I couldn't make it work in the same way for this kind of event.
If you are aware of C code doing what you need, let me know it will be helpful.

**Comment by rtbo** [comment]
closed by #261
